### DISCUSSION TITLE
Aucunes données

### DISCUSSION THREAD

**Discussion by froumizen** [dominikamann/oekofen-pellematic-compact#88] [Q&A]
Bonjour,

Les capteurs ne me donnent aucunes valeurs. 
Mon HA est installé sur une machine virtuelle d'un serveur Proliant tournant avec Truenas Scale.
J'accède au serveur et HA par l'intermédiaire de Cloudflare.
Je n'arrive pas à trouver ce qui empêche la communication de HA avec ma chaudière Oköfen.
Je dois préciser, il me semble, que cela fonctionnait avant que je monte HA sur le serveur quand il tournait sur un PC dédié.
Je joints le journal de l'erreur.
Merci d'avance

Cette erreur provient d'une intégration personnalisée

Enregistreur: custom_components.oekofen_pellematic_compact
Source: custom_components/oekofen_pellematic_compact/__init__.py:143
intégration: Ökofen Pellematic Compact (documentation, problèmes)
S'est produit pour la première fois: 21:16:09 (1 occurrences)
Dernier enregistrement: 21:16:09

Error reading pellematic data
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/urllib/request.py", line 1344, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/local/lib/python3.12/http/client.py", line 1336, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.12/http/client.py", line 1382, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.12/http/client.py", line 1331, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.12/http/client.py", line 1091, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.12/http/client.py", line 1035, in send
    self.connect()
  File "/usr/local/lib/python3.12/http/client.py", line 1001, in connect
    self.sock = self._create_connection(
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/socket.py", line 853, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.12/socket.py", line 838, in create_connection
    sock.connect(sa)
OSError: [Errno 101] Network unreachable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 143, in async_refresh_api_data
    update_result = await self.fetch_pellematic_data()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 159, in fetch_pellematic_data
    result = await self._hass.async_add_executor_job(fetch_data, self._host)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/oekofen_pellematic_compact/__init__.py", line 173, in fetch_data
    response = urllib.request.urlopen(
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 515, in open
    response = self._open(req, data)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 532, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 1373, in http_open
    return self.do_open(http.client.HTTPConnection, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/request.py", line 1347, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 101] Network unreachable>

**Comment by dominikamann** [comment]
Hi @froumizen,

please write in in English!

`File "/usr/local/lib/python3.12/urllib/request.py", line 1347, in do_open
raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 101] Network unreachable> `

Looks like the URL you entered is invalid or not reachable by your ha server. You can check the URL in browser. 

To use Home Assistant (HA) with your heater, ensure that HA—whether running in Docker or on a virtual machine—is on the same network as the heater. Typically, this means connecting your virtual machine to your local (“Home”) network, where the heater resides. If this setup is not possible, you’ll need to create a network tunnel or similar workaround to bridge the networks. However, please note that configuring this network connectivity falls outside the scope of my integration.

Dominik

**Comment by froumizen** [comment]
My boiler is in the same local network as HA. It doesn't work if I put http://192.168.1.××:4321/password json/all in my browser

**Comment by dominikamann** [comment]
A url with port 4321 is the URL of HA not the heater.

**Comment by froumizen** [comment]
I'm sorry, I don't understand. Isn't 8123 the HA port? Shouldn't you put the Boiler IP?

**Comment by dominikamann** [comment]
So if it is also not working in browser I assume the URL (IP) is wrong or the JSON Interface is not activated. Are there may any hints on the Ökofen JSON-Interface setting screen?

**Comment by froumizen** [comment]
Hello, the json interface is activated. The IP is the one found on the boiler.
Is it possible that HA can't get the data because of cloudflare? To access HA from outside I have a domain name.
HA is hosted on VM by Truenas. I have the impression that there is a communication blockage between them. All my other integrations work perfectly (frigate, google...).
It's certainly not due to your integration, because I used it without problems before my change of hardware, but if you have an idea I'm interested because I don't know where to look anymore.
Thank you

**Comment by Ultchad** [comment]
If you can't access to your boiler's JSON API  `http://[YOUR_BOILER_IP]:4321/[YOUR_BOILER_JSON_PASSWORD]/all` with a browser : on the same network as the boiler and HA: The problem is with the boiler.
Is the JSON API enabled, on the boiler ?
Have you access to the web interface of the boiler `http://[YOUR_BOILER_IP]` ?
Can you access the boiler remotely via the application (android, apple)? At home (Same network) and roaming

**Comment by froumizen** [comment]
I have access to the web interface http://ip_boiler, I also have access to the android application locally or roaming.
No access to the json http://[YOUR_BOILER_IP]:4321/[YOUR_BOILER_JSON_PASSWORD]/all with browser locally.
The JSON is active, I tried to restart the boiler without results.
![vue Chaudiere](https://github.com/user-attachments/assets/87d4bdd4-4782-41ff-9eb6-76dd76d16f15)

**Comment by Ultchad** [comment]
What is the error when you try to access to the JSON API with your browser ?

Can you check on the boiler the JSON port (  4321 is default but it can be changed )

**Comment by froumizen** [comment]
The port is indeed 4321, here is the error message:
![Screenshot_20241113_065834_Chrome-EDIT](https://github.com/user-attachments/assets/54c362d2-c64e-46fb-8f6b-27914c29cccf)

**Comment by Ultchad** [comment]
I'm beginning not to know what to test

If you have on Home-Assistant a Plugin/Module Terminal SSH can you test this command
Each command is at a different level of the web page, and the last one on the web page 
`curl` is the command to get the web page as a browser and the `-i` option says to print the HTTP information of the response.

Command:
```bash
curl -i "http://[YOUR_BOILER_IP]:4321/[YOUR_BOILER_JSON_PASSWORD]/all "
curl -i "http://[YOUR_BOILER_IP]:4321/[YOUR_BOILER_JSON_PASSWORD]"
curl -i "http://[YOUR_BOILER_IP]:4321"
curl -i "http://[YOUR_BOILER_IP]"
```

can you return the results with the original command (Change IP by `1.2.3.4` if you want and the password by `QUERTY` )

**Comment by froumizen** [comment]
Here is the result of the test:
![Capture d’écran (9)](https://github.com/user-attachments/assets/85a7aa52-b40c-41ed-9597-4afdd1646170)
Thank you for your help

**Comment by froumizen** [comment]
It's good it works 👍
Seeing in the test results that the boiler was not accessible, I tried to change the parameters, password = ko then the port = ok, I just went to 4322.
I had been looking for a while.
In any case, thank you for accompanying me and for your responsiveness.
Congratulations for your work, I admire you.
